contracts/networks/osmosis/ibc-transfer/src/contract.rs

```diff
@@ -95,14 +96,16 @@ fn execute_ibc_transfer(
         return Err(ContractError::IbcFeesNotSupported);
     }
 
-    // Save in progress ibc transfer data (recover address and coin) to storage, to be used in sudo handler
-    IN_PROGRESS_IBC_TRANSFER.save(
+    // Save in progress recover address to storage, to be used in sudo handler
+    IN_PROGRESS_RECOVER_ADDRESS.save(
         deps.storage,
-        &InProgressIbcTransfer {
-            recover_address: ibc_info.recover_address, // This address is verified in entry point
-            coin: coin.clone(),
-            channel_id: ibc_info.source_channel.clone(),
-        },
+        &ibc_info.recover_address, // This address is verified in entry point
+    )?;
+
+    // Save in progress channel id to storage, to be used in sudo handler
+    IN_PROGRESS_CHANNEL_ID.save(
+        deps.storage,
+        &ibc_info.source_channel, // This address is verified in entry point
```

Comment:
This comment is incorrect because this is a channel identifier, not an address.

Reply:
good catch, changed